plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/data/FieldData.java

```diff
@@ -0,0 +1,101 @@
+package edu.wpi.first.shuffleboard.plugin.base.data;
+
+import com.google.common.collect.ImmutableMap;
+import edu.wpi.first.shuffleboard.api.data.ComplexData;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FieldData extends ComplexData<FieldData> {
+  private final SimplePose2d robot;
+  private final Map<String, SimplePose2d[]> objects;
+
+  public static class SimplePose2d {
+    private final double x;
+    private final double y;
+    private final double degrees;
+
+    @SuppressWarnings("JavadocMethod")
+    public SimplePose2d(double x, double y, double degrees) {
+      this.x = x;
+      this.y = y;
+      this.degrees = degrees;
+    }
+
+    @SuppressWarnings("JavadocMethod")
+    public SimplePose2d(double[] data) {
+      this.x = data[0];
+      this.y = data[1];
+      this.degrees = data[2];
+    }
+
+    public double getX() {
+      return x;
+    }
+
+    public double getY() {
+      return y;
+    }
+
+    public double getDegrees() {
+      return degrees;
+    }
+  }
+
+  public FieldData(SimplePose2d robot, Map<String, SimplePose2d[]> objects) {
+    this.robot = robot;
+    this.objects = objects;
+  }
+
+  @SuppressWarnings("JavadocMethod")
+  public FieldData(Map<String, Object> map) {
+    this.robot = new SimplePose2d((double[]) map.get("Robot"));
+    this.objects = new HashMap<>();
+
+    for (Map.Entry<String, Object> entry : map.entrySet()) {
+      String key = entry.getKey();
+      if (key.equals("Robot") || key.startsWith(".")) {
+        continue;
+      }
+
+      double[] doubles;
+      if (entry.getValue() instanceof byte[]) {
+        byte[] data = (byte[]) entry.getValue();
+        doubles = new double[data.length / Double.BYTES];
+        for (int i = 0; i < doubles.length; i++) {
+          ByteBuffer buffer = ByteBuffer.allocate(Double.BYTES);
+          // this implementation assumes Double.BYTES is 8
+          buffer.put(new byte[]{
```

Comment:
You can use the 3-arg overload of ByteBuffer#put here. Alternatively, ByteBuffer#wrap would probably be the right move.

Reply:
Thanks! That does seem more elegant 😀